### DISCUSSION TITLE
What are the retry strategies for Milo?

### DISCUSSION THREAD

**Discussion by emanodame** [eclipse-milo/milo#1103] [General]
Does Milo implement any retry strategies? I see there is:

https://github.com/eclipse/milo/pull/345 for ClosedChannelException but what does this cover? Any type of request attempt?

**Comment by kevinherron** [comment]
Once connected, the client will automatically reconnect when the underlying channel is lost as well as re-establish a session when the session is lost.

The only strategy involved is an exponential backoff in the delay between attempts, until a 16 second maximum.

There is no retry of any kind of request you might try to execute with the client nor is there a plan to implement any as part of the SDK.
